### ISSUE TITLE
d3d9 driver puts RA in unusuable state

### ISSUE THREAD

**Issue by mick-p1982** [libretro/RetroArch#7588]
## Description

FYI, choosing d3d9 on Windows 10 (Intel chipset) makes the icons as large as the entire screen, and obscures the text, so that it's quite difficult to undo the setting. It didn't work with the core (Beetle) either, but that may be normal.

### Expected behavior

Disable d3d9 option if nonviable.

### Actual behavior

Driver d3d9 falls flat on face.

### Version/Commit

1.7.5

### Environment information

Windows 10

**Comment by hizzlekizzle** [comment]
No surprise with Beetle-PSX-HW, though software rendering should work fine. This is the first I've heard of d3d9 having a problem, though. In fact, we recommend it to anyone with an Intel GPU that's Sandy Bridge (HD 3000) or older...

**Comment by mick-p1982** [comment]
Yeah, it's just the sort of thing that can fall by the wayside. My workstation is fairly recent, integrated i5 with Iris Pro chipset, 5000 or so. It's not ancient hardware or drivers.

(I am in the process of porting a PlayStation game, with many different modes, using the emulator for comparison, and so thought, that using the same driver as in my work could make for fewer surprises. D3D9 is by no means legacy. Outside of commercial wizardry it's the most prolific version of Direct3D.)

**Comment by hizzlekizzle** [comment]
Ok, [I can't reproduce this](https://i.imgur.com/nisM64H.png). The fancy wave shader is gone from the background and the font is a little weird, but icons are the proper size, etc., and Beetle-PSX-HW falls back to software as intended rather than crashing. This is on Win10 with AMD GPU.

**Comment by mick-p1982** [comment]
I'm happy to provide a screenshot (although it's a pain to reverse the setting) but just FYI, you have to close out RA after changing the setting before it will take effect. (Which is kind of a defect in itself.)

**Comment by mick-p1982** [comment]
I think Win32 has an API for closing a program, and automatically restarting it (since that's quite complicated to set up from inside of a program.) If not, I'm thinking about a system restart API. 

I would look at code that has to do with the dimensions of the menu icons. See if there is any code that limits it to a maximum... like the size of the window (quite a huge/unfriendly maximum.) If so, something might be causing the size to be "infinity" and then it is cropped down the the size of the window. It's a lead to go on. (Otherwise, just an OK prompt after change/leaving the menu item can communicate that the change is not yet in effect. I mean ideally, you wouldn't notice the menu change by changing the video driver, assuming the user even understands the setting has anything to do with the menu system.)

How are the icons rendered? And how could they get centered/enlarged to window size. There can't be a lot of code that can do that. If it's not obvious odds are some code is writing into a bad pointer.
